src/components/hooks/useActivityStream.js

```diff
@@ -1,5 +1,30 @@
-import {useState, useEffect} from 'react';
-import {merge} from 'rxjs';
+import {useEffect, useReducer} from 'react';
+
+const APPEND_ACTIVITIES = 'append_activities';
+
+/**
+ * Returns a new array of activityIDs based on the given action.
+ * If no action is passed, it will return the same activityIDs
+ * array without any changes.
+ *
+ * @param {Array.<string>} activityIDs activityIDs associated to the room
+ * @param {obj} action action to apply to given activityIDs
```

Comment:
This should not be `obj` (also in line 33). We should do a quick [typedef|https://jsdoc.app/tags-typedef.html] for the Action object expected.

Line 33 should be a RoomAdapter object type.

Reply:
`typedef` to me is not necessary right now. I did change line 33 to RoomAdapter though